src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -246,44 +365,78 @@ THREE.WebGLShadowMap = function ( _renderer, _lights, _objects ) {
 
 		}
 
+		_renderer.setViewport( _vector4.x, _vector4.y, _vector4.z, _vector4.w );
+
 		_renderer.resetGLState();
 
 		scope.needsUpdate = false;
 
 	};
 
-	function getDepthMaterial( object, material ) {
+	function getDepthMaterial( object, material, isPointLight, lightPositionWorld ) {
 
 		var geometry = object.geometry;
 
 		var useMorphing = geometry.morphTargets !== undefined && geometry.morphTargets.length > 0 && material.morphTargets;
 		var useSkinning = object instanceof THREE.SkinnedMesh && material.skinning;
 
-		var depthMaterial;
+		var newMaterial;
+
+		var depthMaterial = _depthMaterial;
+		var depthMaterialMorph = _depthMaterialMorph;
+		var depthMaterialSkin = _depthMaterialSkin;
+		var depthMaterialMorphSkin = _depthMaterialMorphSkin;
 
-		if ( object.customDepthMaterial ) {
+		if ( isPointLight ) {
+
+			depthMaterial = _distanceMaterial;
+			depthMaterialMorph = _distanceMaterialMorph;
+			depthMaterialSkin = _distanceMaterialSkin;
+			depthMaterialMorphSkin = _distanceMaterialMorphSkin;
+
+		}
 
-			depthMaterial = object.customDepthMaterial;
+		if ( object.customDepthMaterial || object.customDistanceMaterial ) {
+
+			if ( isPointLight ) {
+
+				newMaterial = object.customDistanceMaterial;
+
+			} else {
+
+				newMaterial = object.customDepthMaterial;
+
+			}
```

Comment:
Looking at a subtle issue, here: The presence of either one of the properties may cause the other one, possibly `undefined`, to be assigned to `newMaterial`.

```javascript
var customMaterial = isPointLight ?
        object.customDistanceMaterial :
        object.customDepthMaterial;

if ( customMaterial !== undefined ) {

    newMaterial = customMaterial;

}
```